protocol/0062-SPAM-spam_protection.md

```diff
@@ -83,6 +83,41 @@ Further, each party is allowed to submit up to `n` transactions per epoch where
 
 **Note** `spam.protection.max.updatePartyProfile` must be an integer greater than or equal to `0` (and default to `5`).
 
+### Transaction Spam
+
+#### Derivative markets
+
+Before any order or liquidity commitment is accepted for a perpetual futures or expiring futures check that the party has `margin + order margin + general > 0` with `margin` being the balance in the margin account for the relevant, `order margin` is the balance for the order margin if party is in isolated margin mode for the for the relevant market and `general` the balance in the general account for the relevant asset. Orders from parties that don't meet this criteria are rejected. This is to be done after the PoW check.
+
+Further, given the network parameter: `spam.order.minimalMarginQuantumMultiple` (between 0 and infinite)
+
+If the maintenance margin for a given transaction is smaller than the parameter `spam.order.minimalMarginQuantumMultiple`, then the transaction is pre-block rejected.
```

Comment:
So there will be cases where `margin + order margin`  could be 0, but the transaction should still be accepted. I'm thinking, especially, about cases where a party has a general balance of 0, and margin balance < initial margin, but >= maintenance level (ie they're close to getting closed out). They might want to submit a transaction closing their current position. In that case the maintenance level for the given transaction == 0.

As per an exceptionally long thread on slack, there are also cases, specifically when dealing with capped futures, where the margin requirement for an order can be 0. This will, most likely cause such orders to be rejected as a rule.

To this end, I think it makes more sense for the markets themselves to receive a call to a function like `OrderAcceptanceLikelihood`, which the order spam component keeps track of over time. Should a given party submit N orders that are below 50% likely to get accepted over the course of N blocks, any order transaction that is less than 50% likely to get accepted should get rejected.

The benefits are that individual markets can determine whether or not, in the context of that market, an order makes more or less sense, and therefore should or should not be accepted, without relying on a single network parameter. Because we can keep a tally of these transaction types over time, we could also adjust the threshold for rejection/acceptance of any given transaction on a per-party basis, based on their behaviour. A party that submits, say, a zero-margin order to a capped market once isn't a worry as such. A party doing so every block, or several times per block, may see their order acceptance threshold increased to 75%, or even 100% (effectively timing them out for N blocks).